code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/ServiceDeploymentPlanner.java

```diff
@@ -28,7 +28,7 @@ public ServiceDeploymentPlanner(List<Service> services, List<DeploymentUnit> uni
 
         if (units != null) {
             for (DeploymentUnit unit : units) {
-                if (unit.isUnhealthy()) {
+                if (unit.isUnhealthy() || !unit.isHostActive()) {
```

Comment:
I think I'm good going either way.  The reason I did this was because it looked like "unit unhealthy" is based on the health_state on the instance which I'm assuming is based on the health check specified by the user.  So I didn't want to mix the two in case we want to handle host deletion in a special way.